modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/GeneratePathStructureForEndpoints.testcase

```diff
@@ -0,0 +1,301 @@
+@component-name = "portal-headless"
+definition {
+
+	property custom.properties = "feature.flag.LPS-184413=true${line.separator}feature.flag.LPS-167253=true${line.separator}feature.flag.LPS-178642=true${line.separator}feature.flag.LPS-186757=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless Builder";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginUI();
+	}
+
+	tearDown {
+		ApplicationAPI.deleteAllAPIApplication();
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 4
+	test APIUrlIsPresentInAPIExplorerRestApplications {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("Given an published API application") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "app-test",
+				status = "published",
+				title = "App-test");
+		}
+
+		task ("When I go to API Explorer") {
+			APIExplorer.navigateToOpenAPI();
+		}
+
+		task ("Then the c/{url} is present in Rest Applications") {
+			AssertElementPresent(
+				locator1 = "Button#BUTTON_WITH_VALUE",
+				value = "c/app-test");
+		}
+	}
+
+	@priority = 3
+	test APIUrlsArePresentInAPIExplorer {
```

Comment:
just a reminder that test with priority fp3 are not required in the first phase of automation :)